ratis-common/src/main/java/org/apache/ratis/util/ReferenceCountedLeakDetector.java

```diff
@@ -202,7 +203,7 @@ public synchronized boolean release() {
       }
 
       if (released) {
-        Preconditions.assertNotNull(removeMethod, () -> "Not yet retained (removeMethod == null): " + valueClass);
+        Objects.requireNonNull(removeMethod, "Not yet retained (removeMethod == null): " + valueClass);
```

Comment:
We should keep using the Supplier in order to avoid unnecessarily creating the string.
```java
Objects.requireNonNull(removeMethod, () -> "Not yet retained (removeMethod == null): " + valueClass);
```

Reply:
Thanks @szetszwo  for the comment and review.
I have updated it.